src/node/rpc/node_frontend.h

```diff
@@ -958,6 +964,75 @@ namespace ccf
           "status", ccf::endpoints::OptionalParameter)
         .install();
 
+      auto get_removable_nodes = [this](auto& args, nlohmann::json&&) {
+        GetNodes::Out out;
+
+        auto nodes = args.tx.ro(this->network.nodes);
+        nodes->foreach(
+          [this, &out, nodes](const NodeId& node_id, const NodeInfo& ni) {
+            auto node = nodes->get_globally_committed(node_id);
+            if (
+              node.has_value() && node->status == ccf::NodeStatus::RETIRED &&
+              node->retired_committed)
+            {
+              out.nodes.push_back(
+                {node_id,
+                 node->status,
+                 false /* is_primary */,
+                 node->rpc_interfaces,
+                 node->node_data,
+                 nodes->get_version_of_previous_write(node_id).value_or(0)});
+            }
+            return true;
+          });
+
+        return make_success(out);
+      };
+
+      make_read_only_endpoint(
+        "/network/removable_nodes",
+        HTTP_GET,
+        json_read_only_adapter(get_removable_nodes),
+        no_auth_required)
+        .set_auto_schema<void, GetNodes::Out>()
+        .install();
+
+      auto delete_removed_node = [this](auto& args, nlohmann::json&&) {
+        GetNodes::Out out;
+
+        std::string node_id;
+        std::string error;
+        if (!get_path_param(
+              args.rpc_ctx->get_request_path_params(),
+              "node_id",
+              node_id,
+              error))
+        {
+          return make_error(
+            HTTP_STATUS_BAD_REQUEST, ccf::errors::InvalidResourceName, error);
+        }
+
+        auto nodes = args.tx.rw(this->network.nodes);
+
+        auto node = nodes->get_globally_committed(node_id);
+        if (
+          node.has_value() && node->status == ccf::NodeStatus::RETIRED &&
+          node->retired_committed)
+        {
+          nodes->remove(node_id);
+        }
+
+        return make_success(true);
+      };
+
+      make_endpoint(
+        "/network/nodes/{node_id}",
```

Comment:
This looks like once can remove any node, but this is really about retired nodes. Maybe `GET /network/nodes/retired/{node_id}` instead?

Reply:
It seemed more resty to `DELETE` the same path that you `GET`? One of the attributes is indeed "Retired" for this to work, but that's just a piece of validation?

I don't feel super strongly about this, and I hope to remove this eventually, so I could be swayed either way. @eddyashton @MahatiC?

Reply:
Will stay with the current path naming, as discussed with @eddyashton and @jumaffre.